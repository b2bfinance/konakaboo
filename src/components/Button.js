import styled, { css } from 'styled-components';

const Button = styled.a`
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  font-weight: ${props => props.theme.mainBoldFontWeight};
  justify-content: center;
  padding: 0.75rem 1.5rem;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  user-select: none;

  ${props =>
    props.primary &&
    css`
      background-color: ${props.theme.applyButtonBackground};
      color: ${props.theme.applyButtonColor};
    `};

  ${props =>
    props.secondary &&
    css`
      background-color: ${props.theme.infoButtonBackground};
      color: ${props.theme.infoButtonColor};
    `};

  ${props =>
    props.slim &&
    css`
      padding: 0.25rem 0.5rem;
      font-size: 90%;
    `};
`;

export default Button;
