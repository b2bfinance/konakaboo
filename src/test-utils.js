import { render } from "@testing-library/react";
import { ThemeProvider } from "my-ui-lib";

const Providers = ({ children }) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>;
};

const renderWithProviders = (ui, options) => render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";
export { renderWithProviders as render };
