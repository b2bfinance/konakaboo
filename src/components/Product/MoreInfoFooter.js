import React from 'react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import ApplyButton from './ApplyButton';
import CloseIcon from '@material-ui/icons/Close';
import styled from 'styled-components';

export const Wrapper = styled(Grid)`
  border-top: 2px solid ${props => props.theme.productBorder};
  padding: 16px;
  margin-top: auto;
  min-height: 85px;
`;

export default ({ link, onClose }) => (
  <Wrapper container alignItems="center" justify="space-between">
    <Grid item xs={5} sm={7} lg={8}>
      <Button onClick={onClose} variant="text" size="small">
        <CloseIcon fontSize="small" /> close
      </Button>
    </Grid>
    <Grid item xs={7} sm={5} lg={4}>
      <ApplyButton variant="extended" color="primary" href={link}>
        Continue
      </ApplyButton>
    </Grid>
  </Wrapper>
);
