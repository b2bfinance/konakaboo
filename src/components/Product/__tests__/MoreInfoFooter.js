import Button from '@material-ui/core/Button';
import { createMount, createRender } from '@material-ui/core/test-utils';
import React from 'react';
import MoreInfoFooter, { Wrapper } from '../MoreInfoFooter';

describe('Wrapper', () => {
  const render = createRender();

  test('renders correctly', () => {
    expect(render(<Wrapper />)).toMatchSnapshot();
  });
});

describe('MoreInfoFooter', () => {
  let Component;
  let wrapper;
  let onClose;

  beforeAll(() => {
    const mount = createMount();
    const product = stubData.products.item;

    onClose = jest.fn();
    Component = <MoreInfoFooter onClose={onClose} link={product.links.apply} />;
    wrapper = mount(Component);
  });

  afterAll(() => {
    wrapper.cleanUp();
  });

  test('renders correctly', () => {
    const render = createRender();

    expect(render(Component)).toMatchSnapshot();
  });

  test('should call onClose when close triggered', () => {
    wrapper.find(Button).simulate('click');
    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
