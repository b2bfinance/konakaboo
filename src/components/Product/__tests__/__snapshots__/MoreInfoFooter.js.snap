// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreInfoFooter renders correctly 1`] = `
.c1 {
  background-color: !important;
  width: 100% !important;
  box-shadow: none !important;
}

.c0 {
  border-top: 2px solid;
  padding: 16px;
  margin-top: auto;
  min-height: 85px;
}

<div
  class="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20 c0"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-5-34 MuiGrid-grid-sm-7-50 MuiGrid-grid-lg-8-79"
  >
    <button
      class="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-sizeSmall-121"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-99"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-127 MuiSvgIcon-fontSizeSmall-134"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
         close
      </span>
    </button>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-7-36 MuiGrid-grid-sm-5-48 MuiGrid-grid-lg-4-75"
  >
    <a
      class="MuiButtonBase-root-124 MuiFab-root-136 MuiFab-extended-140 MuiFab-primary-138 c1"
      href="https://b2bfinance.com/enquire/buy-to-let"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiFab-label-137"
      >
        Continue
      </span>
    </a>
  </div>
</div>
`;

exports[`Wrapper renders correctly 1`] = `
.c0 {
  border-top: 2px solid;
  padding: 16px;
  margin-top: auto;
  min-height: 85px;
}

<div
  class="c0"
/>
`;
